src/internal.c

```diff
@@ -1781,6 +1814,12 @@ int InitSSL_Ctx(WOLFSSL_CTX* ctx, WOLFSSL_METHOD* method, void* heap)
 #endif /* HAVE_EXTENDED_MASTER && !NO_WOLFSSL_CLIENT */
 
 #if defined(HAVE_SESSION_TICKET) && !defined(NO_WOLFSSL_SERVER)
+#ifndef WOLFSSL_NO_DEF_TICKET_ENC_CB
+    ret = TicketEncCbCtx_Init(ctx, &ctx->ticketKeyCtx);
```

Comment:
There is nothing dynamic about the data created except in the random.
Generating the name and keys before at context creation, when otherwise not busy, was seen as a better option.
Would you like me to change this?

Reply:
I'm not sure we can make assumptions about whether the rest of the system is more or less busy during ctx init vs first use.  But I do know that seeding the RNG and generating some random data that may not be used takes cycles.

So yes, can we defer until first use if the complexity doesn't increase too much?

Reply:
OK I would prefer seeding to happen up front but you want to avoid the work until necessary.
I'll change the code to not initialise in context creation.

Reply:
Updated to initialize RNG, generate name, generate primary key and set expiry on first call.

Reply:
Thanks.